src/internal/connector/onedrive/excludes/excludes.go

```diff
@@ -1,22 +1,80 @@
 package excludes
 
-import "golang.org/x/exp/maps"
+import (
+	"strings"
+
+	"golang.org/x/exp/maps"
+
+	"github.com/alcionai/corso/src/internal/common/prefixmatcher"
+)
+
+var _ prefixmatcher.MapBuilder = &ParentsItems{}
 
 // Items that should be excluded when sourcing data from the base backup.
 // Parent Path -> item ID -> {}
-type ParentsItems map[string]map[string]struct{}
+type ParentsItems struct {
+	m map[string]map[string]struct{}
```

Comment:
I believe it should be possible to just use the `Builder` interface from the prefixmatcher package. iirc, maps are passed by reference so a `Get` for a key and then copying into the returned map should allow equivalent behavior. If the `Get` returns nothing then a new entry should be added

Reply:
sure, why not